### ISSUE TITLE
Data is normalized multiple times when scale=True in train_predict due to additional transformations in ProcessInputData

### ISSUE THREAD

**Issue by SergejRuff** [Fry2024/OmicsTweezer#3]
When using the high‑level API train_predict(..., scale=True), gene expression data is normalized multiple times, which is likely unintended and can distort input distributions and affect model performance.

```
from OmicsTweezer import train_predict

predict = train_predict.train_predict(
    sc_reference,
    target_bulk,
    num=3000,
    scale=True,
    ot_weight=1
)
```
Similar code is shown in the tutorial script you provided on your github page.

Specifically, normalization is applied both in train_predict and again inside ProcessInputData, resulting in double preprocessing of the same data.

Observed Behavior

```
# First normalization (Min‑Max scaling) 
# In train_predict:

scaler = MinMaxScaler()
train_bulk.X = scaler.fit_transform(train_bulk.X)
test_bulk.X = scaler.fit_transform(test_bulk.X)

#Second normalization (log1p transform)
#In ProcessInputData:

train_x = np.log(train_x + 1)
test_x = np.log(test_x + 1)

# Third normalization (StandardScaler by default)
# Also in ProcessInputData (scaler="ss" by default):

ss = StandardScaler()
ss_train_x = ss.fit_transform(train_x.T).T
ss_test_x = ss.fit_transform(test_x.T).T
```

As a result, gene expression values are transformed as:

Raw counts
 → MinMaxScaler
 → log(x + 1)
 → StandardScaler

is that intended or a mistake? 

Thanks.
